src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1618,9 +1622,8 @@ public static string FullTypeName(this IEdmType type)
         /// <param name="type">Reference to the calling object.</param>
         /// <returns>The element type of this references definition.</returns>
         public static IEdmType AsElementType(this IEdmType type)
-        {
-            IEdmCollectionType collectionType = type as IEdmCollectionType;
-            return (collectionType != null) ? collectionType.ElementType.Definition : type;
+        {   
+            return (type.TypeKind == EdmTypeKind.Collection) ? (type as IEdmCollectionType).ElementType.Definition : type;

```

Comment:
>type.TypeKind [](start = 20, length = 13)

if the input "type = null", the previous one can return null,
but now, the method will throw "null reference exception"

Reply:
Good catch. We should add handling for null.

---
In reply to: [492999471](https://github.com/OData/odata.net/pull/1888#discussion_r492999471) [](ancestors = 492999471)